api/api.go

```diff
@@ -806,3 +814,35 @@ func (b *BlockChainAPI) prepareBlockResponse(
 
 	return blockResponse, nil
 }
+
+func (b *BlockChainAPI) getCadenceHeight(
+	blockNumberOrHash *rpc.BlockNumberOrHash,
+) (uint64, error) {
+	if number, ok := blockNumberOrHash.Number(); ok {
+		height := uint64(0)
+		if number >= 0 {
+			var err error
+			height, err = b.blocks.GetCadenceHeight(uint64(number.Int64()))
+			if err != nil {
+				b.logger.Error().Err(err).Msg("failed to get cadence height")
+				return 0, err
+			}
+		}
+		return height, nil
+	} else if hash, ok := blockNumberOrHash.Hash(); ok {
+		block, err := b.blocks.GetByID(hash)
+		if err != nil {
+			b.logger.Error().Err(err).Msg("failed to get block by hash")
+			return 0, err
+		}
+		height, err := b.blocks.GetCadenceHeight(block.Height)
+		if err != nil {
+			b.logger.Error().Err(err).Msg("failed to get cadence height")
+			return 0, err
+		}
+
+		return height, nil
+	} else {
```

Comment:
no need for last else, just return the error in the end

Reply:
Good point, updated in https://github.com/onflow/flow-evm-gateway/pull/192/commits/a218ad249d6ef9d54abd2574eed160315b89498d